x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/crawler_domain_detail/crawler_domain_detail.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+
+import { useParams } from 'react-router-dom';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiButton, EuiPanel, EuiSpacer } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { generateEncodedPath } from '../../../shared/encode_path_params';
+import { EuiButtonTo } from '../../../shared/react_router_helpers';
+import { SEARCH_INDEX_TAB_PATH } from '../../routes';
+import { EnterpriseSearchContentPageTemplate } from '../layout/page_template';
+import { CrawlCustomSettingsFlyout } from '../search_index/crawler/crawl_custom_settings_flyout/crawl_custom_settings_flyout';
+import { CrawlerStatusIndicator } from '../search_index/crawler/crawler_status_indicator/crawler_status_indicator';
+import { CrawlerStatusBanner } from '../search_index/crawler/domain_management/crawler_status_banner';
+import { getDeleteDomainConfirmationMessage } from '../search_index/crawler/utils';
+import { IndexNameLogic } from '../search_index/index_name_logic';
+import { SearchIndexTabId } from '../search_index/search_index';
+import { baseBreadcrumbs } from '../search_indices';
+
+import { CrawlRulesTable } from './crawl_rules_table';
+import { CrawlerDomainDetailLogic } from './crawler_domain_detail_logic';
+import { DeduplicationPanel } from './deduplication_panel/deduplication_panel';
+import { EntryPointsTable } from './entry_points_table';
+import { SitemapsTable } from './sitemaps_table';
+
+export const CrawlerDomainDetail: React.FC = () => {
+  const { domainId } = useParams<{
+    domainId: string;
+  }>();
+
+  const { indexName } = useValues(IndexNameLogic);
+  const crawlerDomainDetailLogic = CrawlerDomainDetailLogic({ domainId });
+  const { domain, dataLoading } = useValues(crawlerDomainDetailLogic);
+  const { fetchDomainData, deleteDomain } = useActions(crawlerDomainDetailLogic);
+
+  useEffect(() => {
+    fetchDomainData(domainId);
+  }, [domainId]);
+
+  const domainUrl = domain?.url ?? '...';
```

Comment:
I guess so? `CrawlerDomain` will always have a URL, the only time when `domain` is `undefined` is when `dataLoading` is true.  The button to delete domains is disabled when `dataLoading` is true. I can clean up.

Reply:
Actually, no, cause we pull `domainId` from the url path, we always have it, this is just for display.  It could say `...` in the dialog popup (which is being replaced in a follow-up PR with a modal), but it will always use the correct domainId in the logic.